sdks/go/pkg/beam/artifact/materialize.go

```diff
@@ -128,6 +128,11 @@ func (a artifact) retrieve(ctx context.Context, dest string) error {
 		return errors.Wrapf(err, "failed to stat %v", filename)
 	}
 
+	err = os.MkdirAll(filepath.Dir(filename), os.ModePerm)
+	if err != nil {
```

Comment:
Nit: could be inlined into the if (and change the = to := ) but it's not worth blocking for.

Reply:
Might as well do it the best way. Fixed.